packages/core/src/components/Overlay/index.tsx

```diff
@@ -41,15 +43,17 @@ export default class Overlay extends React.PureComponent<Props, State> {
 
     /* istanbul ignore next: refs are hard */
     if (current) {
-      const { x, y } = current.getBoundingClientRect() as DOMRect;
-
-      if (x !== this.state.x) {
-        this.setState({ x });
-      }
-
-      if (y !== this.state.y) {
-        this.setState({ y });
-      }
+      this.rafHandle = requestAnimationFrame(() => {
+        // getBoundingClientRect casues a reflow
+        const { x, y } = current.getBoundingClientRect() as DOMRect;
+        if (x !== this.state.x) {
+          this.setState({ x });
+        }
+
+        if (y !== this.state.y) {
+          this.setState({ y });
+        }
```

Comment:
I think it would be a good idea to put these setStates in another raf. This would probably do the trick:

```js
if (x !== this.state.x || y !== this.state.y) {
  this.rafHandle = requestAnimationFrame(() => {
    this.setState({ x, y });
  });
}
```

Reply:
this caused an issue https://github.com/airbnb/lunar/pull/270#discussion_r360734462